WordPress/pattern-directory
# Add Translation of Patterns via GlotPress

This PR ports work from WordPress.com for Pattern Translation to the Pattern Directory.

There are several working parts to this that work together:
 - Patterns are parsed, and static text is extracted from it
 - Those strings are pushed into GlotPress
 - Those strings are pulled from GlotPress and new Patterns are created in a translated state automatically.

Included are some CLI commands used on WordPress.com, which might make little sense on W.org but I've been using to test & debug.

This is still a work in progress, and will need to be squashed on merge.

Testing this inside the Local Development environment is likely to not be viable, due to the specific integrations with translate.wordpress.org


See #223 


Props @lsl, @dd32, @andrewserong, @deBhal, @enejb, @ramonjd, @akirk, @yuliyan



### Screenshots

An example of an "automatically translated" pattern:
<img width="973" alt="Screen Shot 2021-07-01 at 10 57 03 am" src="https://user-images.githubusercontent.com/767313/124049182-433ecb80-da5b-11eb-98a0-970de6cc24a0.png">




### How to test the changes in this Pull Request:

1. Use the CLI commands to test everything works as expected
2. Run the cron tasks on WordPress.org
3. Verify posts created as expected
